crates/ide/src/annotations.rs

```diff
@@ -99,6 +115,26 @@ pub(crate) fn annotations(
                 });
             }
 
+            if config.annotate_enum_variant_references {
+                let mut enum_variants_metadata: Vec<(TextRange, TextSize)> = Vec::with_capacity(ranges_enum_variants.len());
+                for range_enum_variant in ranges_enum_variants.into_iter() {
+                    let (range, offset) = match range_enum_variant {
+                        Some(range) => (range, range.start()),
+                        None => return,
+                    };
```

Comment:
Instead of straigh up returning early here I think it would make more sense to just skip a variant if it didn't resolve to its source. In which case the extra vec allocation isn't needed either.